setup.py

```diff
@@ -121,5 +121,6 @@ def read(*names, **kwargs):
             "sqlfluff = sqlfluff.cli.commands:cli",
         ],
         "diff_cover": ["sqlfluff = sqlfluff.diff_quality_plugin"],
+        "sqlfluff": ["sqlfluff = sqlfluff.core.plugin.lib"],

```

Comment:
SQLFluff is going to provide an implementation of its own plugin API. This allows us to break the circular dependency by having **`pluggy`** locate the default plugin implementation dynamically, rather than using static imports (which was creating the  dependency cycle).

Reply:
mind = 💥 

everything is a plugin basically! I think their example is lacking on that end ^^

Reply:
Exactly! To be fair, there was nothing inherently "wrong" with your code, except that it created a circular dependency.

I have only used pluggy on projects like SQLFluff and diff-quality, which use it a little. Getting started is the hard part. I understand there are larger projects like pytest where the whole thing is built around plug-ins. I bet it's almost easier that way because developers get used to creating small, independent pieces that connect together loosely.